math/smatrix/include/algebra/math/impl/smatrix_matrix.hpp

```diff
@@ -56,8 +56,38 @@ struct actor {
   // Create identity matrix
   template <unsigned int ROWS, unsigned int COLS>
   ALGEBRA_HOST_DEVICE inline matrix_type<ROWS, COLS> identity() {
-    matrix_type<ROWS, COLS> ret = ROOT::Math::SMatrixIdentity();
-    return ret;
+    return matrix_type<ROWS, COLS>(ROOT::Math::SMatrixIdentity());
+  }
+
+  // Set input matrix as zero matrix
+  template <unsigned int ROWS, unsigned int COLS>
+  ALGEBRA_HOST_DEVICE inline void set_zero(matrix_type<ROWS, COLS> &m) const {
+
+    for (unsigned int i = 0; i < ROWS; ++i) {
+      for (unsigned int j = 0; j < COLS; ++j) {
+        m(i, j) = 0;
```

Comment:
Do you mean `m = zero()` instead of running for loop? Maybe.
I thought calling `zero()` would create an additional matrix, which harm the performance

Reply:
I thought there might be something similar to the identity, but you are right, it probably creates a temporary. Maybe default construction and then move or swap, but it seems ```SMatrix``` does not have a move constructor :( Let's leave it then